### DISCUSSION TITLE
Welcome to AppoMobi.Maui.DrawnUi.Demo Discussions!

### DISCUSSION THREAD

**Discussion by taublast** [taublast/AppoMobi.Maui.DrawnUi.Demo#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by thanosprojects** [comment]
Are we going to have flutter-like ui components on top of skia? Replacing default maui components with skia based custom controls would be quite awesome.

**Comment by taublast** [comment]
That's one of the ways to look a it, one of a simple answers could be a yes. 🤭 If flutter is drawing an app on a canvas why can't we do the same. We were just missing a system of drawn Xamarin/MAUI-friendly controls with bindings, animations, gestures..

Now we have a basic range of drawn controls (think widgets) one can subclass and customise at will. For example, switches/sliders included in the demo demonstrate how easy it is to modify them to any look the client would require. This approach gives the freedom to recreate any design without going laggy, not afraid to use shadows anymore, cache those the non-changing layers as bitmaps and forget about them. At the same time it is very time-effective, for many drawn controls you do not have to care about creating platform dependent code anymore. It has also shown to be very easy too transform an existing maui app into a drawn maui app, like replacing `<StackLayout>` by `<SkiaLayout Type="Column">` etc, my bet is there would even come an automated tool for that at some point. 

The engine source should go public very soon, we are talking weeks, after that hopefully all MAUI devs would get a new tool in their pocket. There is of course still much work to be done, this is an early alpha version, but I hope the community would participate.

**Reply by thanosprojects** [reply]
I tried and played with the project and couldn't believe my eyes. It's very very smooth and performant than what we have in .net maui default controls. The collectionview is just wow.

I hope the first version is released asap. I would definitely love to contribute to the repo as I can. Having flutter like widgets library in .net maui is a dream come true. 

Edit: I remember this repo had been displaayed in .net conf on youtube. I'm sure so many folks are keeping eye on it.

**Comment by yurkinh** [comment]
Hi @taublast 
First of all thanks for all your hard work and effort in making drawn controls a reality.
Waiting for the engine source to go public very much.
Have you thought about making (or porting later) a source engine from SkiaSharp to Maui.Graphics (and use Skiasharp under the hood if needed)?
Are there some limitations in functionality compared to Maui.Graphics?

**Comment by taublast** [comment]
Thank you for the kind words. :) 
If I understand correctly Maui.Graphics is an abstraction wrapper that on some platforms/partly uses SkiaSharp. While we need skia to draw we do not really need another wrapper to apply its limitations. But we are inside the MAUI eco-system so it would be great to marry Maui.Graphics with DrawnUI. At first glance we would need to implement ICanvas interface for our drawing context, so one could draw on it using Maui.Graphics. Would say this is totally possible and (I believe) doesn't looks like a hard task. Could be a very interesting move to make, to create such compatibility and let people reuse their work. Would definitely want to create such handshake with Maui.Graphics.

**Comment by bitzi** [comment]
Hi @taublast 
Thank you for your effort on making this library. I can't believe the performance of collection view, is just incredible. Did you use it in production apps? If yes, can you provide a link to the app?

Why don't you make some YouTube videos about this? It could take off, I think the project is very underrated. This could become the "Flutter" version for .Net MAUI ....

**Comment by taublast** [comment]
Hey, thank you! The whole lib just evolved from working on production apps, you could read about it here, with links to some apps in stores: https://taublast.github.io/posts/MauiJuly/ hey and apps videos too.

I'm using drawn cells and other controls in apps not mentioned in the article that i cannot share much due to some specifics, but trying to create demos examples here and there to fulfill that gap. Please do not miss the Sandbox project from inside the lib repo. 

Indeed the lib hard to adopt primarily because lack of docs. When I have small time between food jobs it's either spend it on some funny cases [like this one](https://github.com/taublast/ShadersCarousel) for example or either write docs that no-one might ever read, guess what I choose :)

Will record a video tomorrow for [Monkey Conf 2024](https://www.monkeyconf.es/ ) if there is little interest I might make more.
